src/Sparrow/Json/Parsing/UnmanagedJsonParserHelper.cs

```diff
@@ -138,8 +138,25 @@ public static IEnumerable<BlittableJsonReaderObject> ReadArrayToMemory(JsonOpera
             if (state.CurrentTokenType != JsonParserToken.StartArray)
                 ThrowInvalidJson(peepingTomStream);
 
+            int docsCountOnCachedRenewSession = 0;
+            bool cachedItemsRenew = false;
             while (true)
             {
+                if (docsCountOnCachedRenewSession <= 16 * 1024)
+                {
+
+                    if (cachedItemsRenew)
+                    {
+                        context.CachedProperties = new CachedProperties(context);
+                        ++docsCountOnCachedRenewSession;
```

Comment:
shouldn't we set cachedItemsRenew to false here?

Reply:
Explicitly not. If we have more than 512 properties in the stream, that is enough so we'll be proactive about it.
Because we are paying for the total size. Better to do it this way.
However, I don't like the fact that we are creating a new `CachedProperties` here for each document.
We should probably clear the existing instance to avoid so many allocations

Reply:
In terms of perf, new allocation to CachedProperties is faster then resetting it.
We need for each document to :
```
 foreach (var cachedSort in _cachedSorts)
            {                
                if (cachedSort != null)
                {
                    cachedSort?.Sorting.Clear();
                    cachedSort.FinalCount = 0;
                }
            }
```

(in debug reset takes 1.4 times then new allocation)

Reply:
What happens if you just do `Array.Clear(_cachedSorts)` ?

Reply:
And in release mode?